File: Week 1/Weekly Practice Problems (Week 1)/1572: Matrix Diagonal Sum/elijah.cs
public class Solution
{
    public int DiagonalSum(int[][] mat)
    {
        int sum = 0;

        for (int i = 0; i < mat.Length; i++)
        {
            sum += mat[i][i];

            if (mat.Length - 1 - i != i)
            {
                sum += mat[mat.Length - 1 - i][i];
            }
        }

        return sum;
    }
}

File: Week 6/Weekly Practice Problems (Week 6)/2643: Row With Maximum Ones/elijah.cpp
class Solution {
public:
    vector<int> rowAndMaximumOnes(vector<vector<int>>& mat)
    {
        int maxRow = 0;
        int maxRowCount = 0;

        for (int r = 0; r < mat.size(); r++)
        {
            int count = 0;
            for (int c = 0; c < mat[r].size(); c++)
            {
                count += mat[r][c];
            }

            if (count > maxRowCount)
            {
                maxRowCount = count;
                maxRow = r;
            }
        }

        return vector<int> { maxRow, maxRowCount };
    }
};
